From dc0fdb8024999b1ff508f0cef0737bb314e5d6c2 Mon Sep 17 00:00:00 2001
From: Muthukumar K <muthukumar.kandasamy@intel.com>
Date: Thu, 30 Aug 2012 20:31:13 +0530
Subject: [PORT FROM MAIN] Tombstone while doing video streaming when server sends non-standard 90kHz (SRC assertion failure)

BZ: 50674

Feng server sets 90k sample rate for all the streams (both
audio and video). The CHECK here results in a crash as 90k
is not the standard one. The sample rate is recalculated in
the codec with the stream data. Thus removal of this CHECK
does no harm.

DONOT UPSTREAM

Change-Id: I1c9e703071d2b125fa3f5fde30b7d7baccc1ff62
Orig-Change-Id: Icf78cb650c73a04959d9131e5d5754d5b6710e0c
Signed-off-by: Muthukumar K <muthukumar.kandasamy@intel.com>
Reviewed-on: http://android.intel.com:8080/81068
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 .../rtsp/AMPEG4ElementaryAssembler.cpp             |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/rtsp/AMPEG4ElementaryAssembler.cpp b/media/libstagefright/rtsp/AMPEG4ElementaryAssembler.cpp
index 2ace475..ad58c80 100644
--- a/media/libstagefright/rtsp/AMPEG4ElementaryAssembler.cpp
+++ b/media/libstagefright/rtsp/AMPEG4ElementaryAssembler.cpp
@@ -103,6 +103,7 @@ static bool GetSampleRateIndex(int32_t sampleRate, size_t *tableIndex) {
         }
     }
 
+    ALOGW("%d is a non-standarad sample rate.", sampleRate);
     return false;
 }
 
@@ -192,7 +193,9 @@ AMPEG4ElementaryAssembler::AMPEG4ElementaryAssembler(
                 desc.c_str(), &sampleRate, &numChannels);
 
         mChannelConfig = numChannels;
-        CHECK(GetSampleRateIndex(sampleRate, &mSampleRateIndex));
+        // CHECK removed here as SR is recalculated in codec from stream data.
+        // Feng server sets 90k SR default. This removal is necessary then.
+        GetSampleRateIndex(sampleRate, &mSampleRateIndex);
     }
 }
 
-- 
1.7.4.1

